drivers/gpio/gpio_mchp_xec.c

```diff
@@ -213,6 +217,19 @@ static int gpio_xec_pin_interrupt_configure(const struct device *dev,
 	*current_pcr1 = (*current_pcr1 & ~mask) | pcr1;
 
 	if (mode != GPIO_INT_MODE_DISABLED) {
+		mask = 0u;
+		/* HW takes several clocks to stabilize when edge
+		 * detect is enabled.
+		 */
+		for (int i = 0; i < XEC_GPIO_EDGE_DLY_COUNT; i++) {
```

Comment:
This looks indeed like a hardware bug. The other, much less likely cause would be the processor reordering access to device registers. The registers are normally placed in Device / Strongly-ordered memory region to guarantee the transaction order but maybe it's worth giving it a try and replacing the delay with a call to `__DMB()`?

Reply:
@mnkp I tried replacing the loop with __DMB() in the PR and on the MEC172x GPIO driver.  It worked both places.  Updated the PR to use __DMB.

Reply:
Interesting, I actually didn't expect `__DMB()` would work. GPIO and interrupt controller modules should be placed in Device / Strongly-ordered memory region. 

Such erroneous behavior could be caused by misconfigured MPU but MEC15xx EVB has Cortex M4, without MPU. So it does look like a hardware bug. @scottwcpg you work for Microchip, don't you? Could you check the documentation, contact hardware engineers and vefify which module is causing problems. If GPIO / interrupt controller registers are not placed in Device / Strongly-ordered memory region we're likely to have more bugs in the driver's code.

Reply:
@mnkp I was also surprised by __DMB working especially on the MEC172x running at 96MHz.  All the Microchip peripherals are on AHB in the 0x4000_0000 to 0x4FFFF_FFFF range which is the "System Bus" in ARM terminology.  ARM docs state the write buffer is used for DCode and System bus.

Reply:
hi all, see below feedback from Hardware team at mchp.

1. mec52x and mec172x GPIO logic requires 2 synchronization stages + some extra clocks once an event is detected to clear out the internal logic. This will take ~8 system clock cycles (it can take less).
    we automatically disable clocking for gpio block when there’s no activity and then automatically re-enable clocks when the async detectors capture changes on the pin. This has few clocks delay.
2. It will always trigger an edge detect when turned on if the pin state is the level of the end condition (falling edge = pin state 0; rising edge = pin state 1). It will sometimes trigger an edge detect on any GPIO IRQ configuration change.
3.  dmb instruction has no impact as all ARM AHB accesses stall the processor until completed which gives a large delay between adjacent AHB accesses. 
4. above restriction are applicable only for GPIO block in mec152x device and mec172x device. We need to have delay between write to control register and clearing interrupt status register before enabling GPIO edge detection interrupt.

Reply:
Considering the above reply I believe we need to change the way the fix is implemented. Instead of using __DMB instruction go back to a loop which adds a short delay.

Reply:
@mnkp I agree. Since the PR's for both chips have been merged. I will create a new PR updating both drivers to use the delay loop.